wgpu-core/src/instance.rs

```diff
@@ -603,6 +622,20 @@ impl<G: GlobalIdentityHandlerFactory> Global<G> {
                 );
             }
 
+            // Verify all extensions were exposed by the adapter
+            if !adapter.unsafe_extensions.allowed() {
```

Comment:
this isn't needed here, I think. The user can't request extensions that adapter doesn't provide, anyway

Reply:
I know it I isn't strictly necessary, but I did it to try to provide a more helpful error messages.